daml-lf/spec/daml-lf-1.rst

```diff
@@ -1887,140 +1887,123 @@ will always be used to compare values of same types::
+  ———————————————————————————————————————————————————
+    'TypeRep' <ₜ 'Update'
 
-   Mod:T:E₁ comes before Mod:T:E₂ in the list of
-   constructors for enum type Mod:T
-  ——————————————————————————————————————————————————— GenLtEnumCon
-   Mod:T:E₁ <ᵥ Mod:T:E₂
+  ———————————————————————————————————————————————————
+    'Update' <ₜ Mod:T
 
-   v₁ ~ᵥ v₁   ⋯   vᵢ₋₁ ~ᵥ wᵢ₋₁     vᵢ <ᵥ wᵢ    i <= n
-  ——————————————————————————————————————————————————— GenLtStructCon
-   ⟨ f₁ = v₁, …, fₘ = vₘ ⟩ ~ᵥ ⟨ f₁ = w₁, …, fₘ = wₘ ⟩
+    PkgId₁ < PkgId₂
+  ———————————————————————————————————————————————————
+    (PkgId₁:ModName₁):T₁ <ₜ (PkgId₂:ModName₂):T₂
+
+    ModName₁ < ModName₂
+  ———————————————————————————————————————————————————
+    (PkgId:ModName₁):T₁ <ₜ (PkgId:ModName₂):T₂
+
+    T₁ < T₂
+  ———————————————————————————————————————————————————
+    Mod:T₁ <ₜ Mod:T₂
+
+  ———————————————————————————————————————————————————
+    Mod:T <ₜ n
+
+    n₁ is strictly less that n₂
+  ———————————————————————————————————————————————————
+    n₁ <ₜ n₂
+
+  ———————————————————————————————————————————————————
+    n <ₜ ⟨ f₁ : τ₁, …, fₘ : τₘ ⟩
+
+    fᵢ < gᵢ
+  ———————————————————————————————————————————————————
+    ⟨ f₁ : τ₁, …, fₘ : τₘ ⟩ <ₜ
+      ⟨ f₁ : σ₁, …, fᵢ₋1 : σᵢ₋₁, gᵢ : σᵢ, …, gₙ : σₙ ⟩
 
-  ——————————————————————————————————————————————————— GenLtTextMap1
-   [ ] <ᵥ [s₁ ↦ w₁, …, sₘ ↦ wₘ]
-
-   t₁ < t₂ < … < tₘ
-   s₁ < s₂ < … < sₙ
-   t₁ < s₁
-  ——————————————————————————————————————————————————— GenLtTextMap2
-   [t₁ ↦ v₁, …, tₘ ↦ vₘ] <ᵥ [s₁ ↦ w₁, …, sₙ ↦ wₙ]
-
-   t₁ < t₂ < … < tₘ
-   s₁ < s₂ < … < sₙ
-   t₁ = s₁
-   v₁ <ᵥ w₁
-  ——————————————————————————————————————————————————— GenLtTextMap3
-   [t₁ ↦ v₁, …, tₘ ↦ vₘ] <ᵥ [s₁ ↦ w₁, …, sₙ ↦ wₙ]
-
-   t₁ = s₁
-   v₁ ~ᵥ w₁
-   [t₂ ↦ v₂, …, tₘ ↦ vₘ] <ᵥ [s₂ ↦ w₂, …, sₙ ↦ wₙ]
-  ——————————————————————————————————————————————————— GenLtTextMap4
-   [t₁ ↦ v₁, …, tₘ ↦ vₘ] <ᵥ [s₁ ↦ w₁, …, sₙ ↦ wₙ]
-
-  ——————————————————————————————————————————————————— GenLtGenMap1
-   [ ] <ᵥ [s₁ ↦ w₁, …, sₘ ↦ wₘ]
-
-   t₁ <ᵥ t₂ <ᵥ … <ᵥ tₘ
-   s₁ <ᵥ s₂ <ᵥ … <ᵥ sₙ
-   t₁ <ᵥ s₁
-  ——————————————————————————————————————————————————— GenLtGenMap2
-   [t₁ ↦ v₁, …, tₘ ↦ vₘ] <ᵥ [s₁ ↦ w₁, …, sₙ ↦ wₙ]
-
-   t₁ <ᵥ t₂ <ᵥ … <ᵥ tₘ
-   s₁ <ᵥ s₂ <ᵥ … <ᵥ sₙ
-   t₁ ~ᵥ s₁
-   v₁ <ᵥ w₁
-  ——————————————————————————————————————————————————— GenLtGenMap3
-   [t₁ ↦ v₁, …, tₘ ↦ vₘ] <ᵥ [s₁ ↦ w₁, …, sₙ ↦ wₙ]
-
-   t₁ ~ᵥ s₁
-   v₁ ~ᵥ w₁
-   [t₂ ↦ v₂, …, tₘ ↦ vₘ] <ᵥ [s₂ ↦ w₂, …, sₙ ↦ wₙ]
-  ——————————————————————————————————————————————————— GenLtGenMap4
-   [t₁ ↦ v₁, …, tₘ ↦ vₘ] <ᵥ [s₁ ↦ w₁, …, sₙ ↦ wₙ]
-
-.. note: In the above rules, map entries for TextMap and GenMap are ordered
-   by key. The rules make this assumption explicit.
+  ———————————————————————————————————————————————————
+    ⟨ f₁ : τ₁, …, fₘ : τₘ ⟩ <ₜ
+      ⟨ f₁ : τ₁, …, fₘ : τₘ, …, fₙ : σₙ ⟩
 
+    τᵢ <ₜ σᵢ
+  ———————————————————————————————————————————————————
```

Comment:
I think these rules would be significantly easier to understand if we give them somewhat descriptive names or a comment.

Reply:
Good point, though not easy. Will try my best.